cmd/controller-manager/app/controller-manager.go

```diff
@@ -243,20 +243,26 @@ func getKubeFedConfig(opts *options.Options) *corev1b1.KubeFedConfig {
 	return fedConfig
 }
 
-func setDefaultKubeFedConfigScope(fedConfig *corev1b1.KubeFedConfig) {
-	if len(fedConfig.Spec.Scope) == 0 {
-		// TODO(sohankunkerkar) Remove when no longer necessary.
-		// This Environment variable is a temporary addition to support Red Hat's downstream testing efforts.
-		// Its continued existence should not be relied upon.
-		const defaultScopeEnv = "DEFAULT_KUBEFED_SCOPE"
-		defaultScope := os.Getenv(defaultScopeEnv)
-		if len(defaultScope) != 0 {
-			if defaultScope != string(apiextv1b1.ClusterScoped) && defaultScope != string(apiextv1b1.NamespaceScoped) {
-				klog.Fatalf("%s must be Cluster or Namespaced; got %q", defaultScopeEnv, defaultScope)
-			}
+func setDefaultKubeFedConfigScope(fedConfig *corev1b1.KubeFedConfig) bool {
+	const defaultScopeEnv = "DEFAULT_KUBEFED_SCOPE"
+	defaultScope := os.Getenv(defaultScopeEnv)
+	if len(defaultScope) != 0 {
+		if defaultScope != string(apiextv1b1.ClusterScoped) && defaultScope != string(apiextv1b1.NamespaceScoped) {
+			klog.Fatalf("%s must be Cluster or Namespaced; got %q", defaultScopeEnv, defaultScope)
```

Comment:
@font: That is an existing log message, but yes, I will change it. BTW, I think it's ok to catch the error early rather than wait for `ValidateKubeFedConfig` to get called.

Reply:
It's not just waiting for `ValidateKubeFedConfig` (even though it's just as early IMO). This code was implemented at a time when the validation webhook did not exist. So the next function calls to either create or update the KubeFedConfig will trigger a check by the validation webhook anyway. A failure to create or update the KubeFedConfig results in a fatal message as well rendering this check repetitive.